tool/src/main/java/org/wildfly/security/tool/FileSystemEncryptRealmCommand.java

```diff
@@ -0,0 +1,770 @@
+                    count.getAndIncrement();
+                } else {
+                    String[] parts = line.split(":");
+                    String option = parts[0];
+                    String arg = parts[1];
+                    switch (option) {
+                        case INPUT_REALM_LOCATION_PARAM:
+                            descriptor.setInputRealmLocation(arg);
+                            break;
+                        case OUTPUT_REALM_LOCATION_PARAM:
+                            descriptor.setOutputRealmLocation(arg);
+                            break;
+                        case REALM_NAME_PARAM:
+                            descriptor.setFileSystemRealmName(arg);
+                            break;
+                        case CREDENTIAL_STORE_LOCATION_PARAM:
+                            descriptor.setCredentialStore(arg);
+                            break;
+                        case SECRET_KEY_ALIAS_PARAM:
+                            descriptor.setSecretKeyAlias(arg);
+                            break;
+                        case HASH_CHARSET_PARAM:
+                            descriptor.setHashCharset(Charset.forName(arg));
+                            break;
+                        case HASH_ENCODING_PARAM:
+                            descriptor.setHashEncoding(Encoding.valueOf(arg.toUpperCase()));
+                            break;
+                        case LEVELS_PARAM:
+                            descriptor.setLevels(Integer.parseInt(arg));
+                            break;
+                    }
+                }
+            });
+        } catch (IOException e) {
+            errorHandler(e);
+        }
+        int currentCount = count.intValue();
+        findMissingRequiredValuesAndSetValues(currentCount, descriptor);
+        copyAddResetDescriptor(descriptor);
+        if (summaryMode) {
+            printDescriptorBlocks(currentCount);
+        }
+        count.getAndIncrement();
+    }
+
+    /**
+     * Copies a Descriptor into a temporary one for it can be cloned into the list of descriptors
+     *
+     * @param original The original descriptor that is continually modified
+     */
+    private void copyAddResetDescriptor(Descriptor original) {
+        Descriptor temp = new Descriptor(original);
+        descriptors.add(temp);
+        original.reset();
+    }
+
+    /**
+     * Determines if the current descriptor block is missing any required values
+     * and then if it is, prints out a warning message and sets that required value
+     * and the optional values to null
+     *
+     * @param count The index of the current descriptor block
+     */
+    private void findMissingRequiredValuesAndSetValues(int count, Descriptor descriptor) {
+        boolean missingRequiredValue = false;
+        if (descriptor.getInputRealmLocation() == null) {
+            warningHandler(ElytronToolMessages.msg.skippingDescriptorBlock(count, "missing input realm location"));
```

Comment:
It would be better to incorporate the reason into the ElytronToolMessages so all the text is there.

Reply:
Added